lib/rules/prop-types.js

```diff
@@ -301,6 +301,17 @@ module.exports = {
     }
 
     /**
+     * Removes quotes from around an identifier.
+     * @param {string} the identifier to strip
+     */
+    function stripQuotes(string) {
+      if (string[0] === '\'' || string[0] === '"') {
```

Comment:
@ljharb, well, it took a bit more than a weekend, but i've updated the function as you suggest.